fluent/trigger-conf.js

```diff
@@ -130,7 +171,115 @@ class ItemTriggerConfig {
     }
 }
 
+class ThingTriggerConfig {
+    constructor(thingUID) {
+        this.thingUID = thingUID;
+    }
+
+    _complete() {
+        return typeof (this.op_type) !== 'undefined';
+    }
+
+    describe(compact) {
+        switch (this.op_type) {
+            case "changed":
+                let transition = 'changed';
+
+                if (this.to_value) {
+                    transition += ` to ${this.to_value}`;
+                }
+
+                if (this.from_value) {
+                    transition += ` from ${this.from_value}`;
+                }
+
+                return `${this.thingUID} ${transition}`;
+            case "updated":
+                return compact ? `${this.thingUID}/updated` : `Thing ${this.thingUID} received update`;
+            default:
+                throw error("Unknown operation type: " + this.op_type);
+        }
+    }
+
+    changed() {
+        this.op_type = 'changed';
+        return this;
+    }
+
+    updated() {
+        this.op_type = 'updated';
+        return this;
+    }
+
+    from(value) {
+        if (this.op_type != 'changed') {
+            throw ".from(..) only available for .changed()";
+        }
+        this.from_value = value;
+        return this;
+    }
+
+    to(value) {
+        this.to_value = value;
+        return this;
+    }
+
+    _toOHTriggers() {
+        switch (this.op_type) {
+            case "changed":
+                return [triggers.ThingStatusChangeTrigger(this.thingUID, this.to_value, this.from_value)];
+            case 'updated':
+                return [triggers.ThingStatusUpdateTrigger(this.thingUID, this.to_value)]
+            default:
+                throw error("Unknown operation type: " + this.op_type);
+        }
+    }
+};
+
+class SystemTriggerConfig {
+    constructor() {
+        this._toOHTriggers = () => [triggers.SystemStartlevelTrigger(this.level)]
+        this.describe = (compact) => compact ? `system:${this.level}` : `system level "${this.level}"`
+    }
+    _complete() {
+        return typeof (this.level) !== 'undefined';
+    }
+
+    rulesLoaded() {
+        this.level = 40;
```

Comment:
I should probably just call `return this.startLevel(40)` for this and the other methods